iep-012/README.adoc

```diff
@@ -0,0 +1,198 @@
+
+The big picture is to create an environment charts, one to rules them all.
+This main chart is responsible to deploy jenkins credentials as Kubernetes secret, set a custom jenkins image with specific configuration and finally some chart dependencies to deploy third services (Artifactory,maven registry, ...)
+
+
+For example, let's imagine two different environments, security and release.
+The security environment generate short-live artifacts and so is configured to deploy every services needed to share those artifacts (maven, docker, war, etc.). It's easier to control the access, we don't pollute the official repository.
+But in the case of the release environment, it doesn't make sense to deploy a maven repository behind maven.release.jenkins.io as we already have link:https://repo.jenkins-ci.org[repo.jenkins-ci.org]
+So we just map http://repo with https://repo.jenkins-ci.org.
+
+.Environment
+image:images/environment_namespace.png[]
+
+Using Kubernetes as CI/CD environment is not only about how we deploy and configure Jenkins but how we deploy and configure all third services that we need.
+
+This repository use Jenkins-x to configure an environment https://github.com/olblak/jenkins-release-environment. Jenkins-x is an opinionated way for configuring such environment on Kubernetes.
+
+Following table display the different endpoints that we can use for each environment where the internal endpoint are the same for every environment.
+
+[cols="h,3*", options="header"]
+.Endpoint
+|===
+|       |  Repo | Jenkins | Registry
+| Internal | http://repo | http://ci | http://registry
+| Weekly | http://repo.weekly.jenkins.io | https://ci.weekly.jenkins.io | http://registry.weekly.jenkins.io
+| Security | https://repo.cert.jenkins.io | https://ci.cert.jenkins.io | https://ci.registry.jenkins.io
+| Release | https://repo.jenkins-ci.org | https://ci.release.jenkins.io | https://hub.docker.com
+|===
+
+=== Workflow
+It's now time to cover the last topic, which is how we build, test and release a specific version for an environment. Jenkins is used as the conductor by means of Jenkinsfile and will execute a succession of actions defined link:https://github.com/olblak/jenkins/blob/master/Jenkinsfile.release[here].
+In order to simplify the maintainability of this script, most of the logic as been moved to a shell link:https://github.com/olblak/jenkins/blob/master/scripts/buildJenkins.bash[script] that accept various arguments.
+
+All endpoints are hard-coded with Kubernetes internal endpoint like "repo", jenkins, ... And any other configuration is done via environment variables configured at the environment level.
+So every environment use the same Jenkinsfile but the output change depending on the environment configuration and the git repository.
+
+Maven with the link:http://maven.apache.org/maven-release/maven-release-plugin/[release plugin] is used to build Jenkins(link:https://wiki.jenkins.io/pages/viewpage.action?pageId=3309681[why?]).
```

Comment:
Well that page is about using Maven generally, not `maven-release-plugin` specifically.